routing/pathfind.go

```diff
@@ -483,14 +478,25 @@ func findPath(g *graphParams, r *RestrictParams, source, target route.Vertex,
 		// map is populated with this edge.
 		distance[fromVertex] = nodeWithDist{
 			dist:            tempDist,
-			node:            fromNode,
+			node:            fromVertex,
 			amountToReceive: amountToReceive,
 			fee:             fee,
 			incomingCltv:    incomingCltv,
 		}
 
 		next[fromVertex] = edge
 
+		// If fromVertex exists in the heap, then we must call heap.Fix
+		// to adjust its position in the heap. Otherwise, we will end up
+		// performing more db lookups on the same node.
+		if index, ok := nodeHeap.pubkeyIndices[fromVertex]; ok {
+			// Change the value at the specified index.
+			nodeHeap.nodes[index] = distance[fromVertex]
+			// Call heap.Fix to reorder the heap.
+			heap.Fix(&nodeHeap, index)
+			return
+		}
```

Comment:
This could go into a method of `distanceHeap`?

Reply:
Done